### PR TITLE
Feature/us5498 fund - Merge to get Fund Component Into Dev 

### PR DESCRIPTION
**Pull request by vchtelmakh**
This is not a full feature. Will continue working from this branch.

### CODE REVIEW

File: src/app/fund-and-frequency/fund-and-frequency.component.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { GiftService } from '../services/gift.service';
+import { Router } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
+
```
Comment (mjvcallibrity) [inline src/app/fund-and-frequency/fund-and-frequency.component.ts]: Please group imports by application imports vs 3rd party imports. Please group alphabetically by module. Please group destructured imports alphabetically. For any questions, see the Angular 2 style guide, [Style 03-06](https://angular.io/docs/ts/latest/guide/style-guide.html#03-06)

File: src/app/fund-and-frequency/fund-and-frequency.component.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  availableFrequencies: Array<string> = this.defaultFrequencies;
+  form: FormGroup;
+  startDate: any;
+  fundIdParam: number;
+  defaultFund: Program = {
+    'ProgramId': 3,
+    'Name': 'General Giving',
+    'ProgramType': 1,
+    'AllowRecurringGiving': true
+  };
```
Comment (mjvcallibrity) [inline src/app/fund-and-frequency/fund-and-frequency.component.ts]: Please alphabetize member sequence. [Style 05-14](https://angular.io/docs/ts/latest/guide/style-guide.html#05-14)

File: src/app/fund-and-frequency/fund-and-frequency.component.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if (frequency === 'One Time') {
+      this.resetDate();
+    }
+  }
+
+  onClickDate(newValue: any) {
+    this.gift.start_date = newValue;
+  }
+
+}
```
Comment (mjvcallibrity) [inline src/app/fund-and-frequency/fund-and-frequency.component.ts]: Please alphabetize methods. [Style 05-14](https://angular.io/docs/ts/latest/guide/style-guide.html#05-14)

File: src/app/services/gift.service.ts [hunk 4]
```diff
@@ -56,6 +56,11 @@ export class GiftService {
   public cvv: string = '';
   public zipCode: string = '';
 
+  //Fund and frequency information
+  public fund: any = '';
+  public start_date: any = '';
+  public frequency: any = '';
+
```
Comment (mjvcallibrity) [inline src/app/services/gift.service.ts]: Please alphabetize. [Style 05-14](https://angular.io/docs/ts/latest/guide/style-guide.html#05-14)

File: src/app/fund-and-frequency/fund-and-frequency.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+import { StateManagerService } from '../services/state-manager.service';
+import { PreviousGiftAmountService } from '../services/previous-gift-amount.service';
+import { LoginService } from '../services/login.service';
+import { ExistingPaymentInfoService } from '../services/existing-payment-info.service';
+import { CookieService } from 'angular2-cookie/services/cookies.service';
+import { HttpClientService } from '../services/http-client.service';
+import { PaymentService } from '../services/payment.service';
+import { StripeService } from '../services/stripe.service';
+
```
Comment (mjvcallibrity) [inline src/app/fund-and-frequency/fund-and-frequency.component.spec.ts]: Please group imports by application imports vs 3rd party imports. Please group alphabetically by module. Please group destructured imports alphabetically. For any questions, see the Angular 2 style guide, [Style 03-06](https://angular.io/docs/ts/latest/guide/style-guide.html#03-06)

File: src/app/app.module.ts [hunk 6]
```diff
@@ -16,6 +16,7 @@ import { PreloaderModule } from './preloader/preloader.module';
 import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
 import { PaymentComponent } from './payment/payment.component';
 import { DonationComponent } from './donation/donation.component';
+import { FundAndFrequencyComponent  } from './fund-and-frequency/fund-and-frequency.component';
```
Comment (mjvcallibrity) [inline src/app/app.module.ts]: Please group imports by application imports vs 3rd party imports. Please group alphabetically by module. Please group destructured imports alphabetically. For any questions, see the Angular 2 style guide, [Style 03-06](https://angular.io/docs/ts/latest/guide/style-guide.html#03-06)

File: src/app/fund-and-frequency/fund-and-frequency.component.ts (hunk 2 above)
Comment (vchtelmakh) [inline src/app/fund-and-frequency/fund-and-frequency.component.ts]: @richardbarker Rick!!

**Pull request merged**